engine-tests/src/test/java/org/terasology/engine/persistence/serializers/TypeSerializerTest.java

```diff
@@ -85,14 +95,50 @@ void testDeserialize() {
     }
 
     @Test
-    void testSerializeDeserialize() throws IOException {
+    void testProtobufSerializeDeserialize() throws IOException {
         byte[] bytes = protobufSerializer.toBytes(INSTANCE, new TypeInfo<SomeClass<Integer>>() {
         });
 
         SomeClass<Integer> deserializedInstance =
                 protobufSerializer.fromBytes(bytes, new TypeInfo<SomeClass<Integer>>() {
                 });
 
+        assertNotEmpty(typeHandlerLibrary.getTypeHandler(Animal.class));
+        assertNotEmpty(typeHandlerLibrary.getTypeHandler(Dog.class));
+
+        assertNotEmpty(typeRegistry.load("org.terasology.engine.persistence.serializers.TypeSerializerTest$Animal"));
+        assertNotEmpty(typeRegistry.load("org.terasology.engine.persistence.serializers.TypeSerializerTest$Dog"));
+
+        ModuleEnvironment env = moduleManager.getEnvironment();
+        assertNotEmpty(env.getSubtypesOf(Animal.class));
+
+//        assertNotEmpty(typeRegistry.getSubtypesOf(Animal.class));
+
+        assertEquals(INSTANCE, deserializedInstance);
+    }
+
+    @Test
+    void testJsonSerializeDeserialize() throws IOException {
+        //noinspection unchecked,OptionalGetWithoutIsPresent,OptionalGetWithoutIsPresent
+        byte[] bytes = (byte[]) gsonSerializer.serialize(INSTANCE, new TypeInfo<SomeClass<Integer>>() {
+        }).get();
+
+        //noinspection unchecked,OptionalGetWithoutIsPresent,OptionalGetWithoutIsPresent
+        SomeClass<Integer> deserializedInstance =
+                (SomeClass<Integer>) gsonSerializer.deserialize(new TypeInfo<SomeClass<Integer>>() {}, bytes)
+                .get();
+
+        assertNotEmpty(typeHandlerLibrary.getTypeHandler(Animal.class));
+        assertNotEmpty(typeHandlerLibrary.getTypeHandler(Dog.class));
+
+        assertNotEmpty(typeRegistry.load("org.terasology.engine.persistence.serializers.TypeSerializerTest$Animal"));
+        assertNotEmpty(typeRegistry.load("org.terasology.engine.persistence.serializers.TypeSerializerTest$Dog"));
+
+//        ModuleEnvironment env = moduleManager.getEnvironment();
+//        assertNotEmpty(env.getSubtypesOf(Animal.class));
+
+//        assertNotEmpty(typeRegistry.getSubtypesOf(Animal.class));
```

Comment:
FIXME: bunch o clutter added to these tests.

should def remove the commented-code. but should the extra assertions stay? or are they out of scope for the class under test?